packages/runtime/container-runtime/src/summarizer.ts

```diff
@@ -387,36 +389,33 @@ export class RunningSummarizer implements IDisposable {
         }
     }
 
-    private trySummarize(reason: string): { broadcastP: Promise<void> } {
-        if (this.summarizing === true) {
+    private trySummarize(reason: string): void {
+        if (this.summarizing !== undefined) {
             // We can't summarize if we are already
             this.tryWhileSummarizing = true;
-            return { broadcastP: Promise.resolve() };
+            return;
```

Comment:
It feels to me that you can make this.summarizing be scoped only to this function by returning this.summarizing.promise here, and thus making code in waitStop() await this.trySummarize() call.